PAINTROID-208 Register intent for .ora files

harsh-hash opened the pull request:
Ticket solve - https://jira.catrob.at/browse/PAINTROID-208
- added intent-filter for .ora files
- added test to check intent

*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

wslany reviewed (changes requested):
Please fix inlined comments and please also fix all lint warnings, see https://jenkins.catrob.at/job/Paintroid/job/PR-855/2/analysis/ 

Also, I tried it with Android 10 on a Galaxy S20 Ultra, and while it opened the ora file when I tried to open it with a file browser from Samsung, the image in Pocket Paint (after force closing Pocket Paint first!) was then _empty_. Trying to open the same ora file with the Load image option in Pocket Paint worked fine, so it was not about the ora file or Pocket Paint's general ability to open the format. On what device and Android version did you test it where it worked, and with which file browser or other app? Please add a test for this.

Another issue: this PR now makes Pocket Paint an option for _any_ file type, which is of course wrong. E.g., even when trying to open an .mp3 or .mp4 file, Pocket Paint is offered, see screenshot for a txt (text) file:
![image](https://user-images.githubusercontent.com/1747888/112958037-554e5c80-9142-11eb-87f9-0e1de72d118b.png)

Please add a test for the latter, namely that it does *not* work, and please develop in a test-driven-way, i.e., test-first.

harsh-hash commented:
Till now I haven't written any code to load the image in the app. I have just added an intent filter for the ora file format and added a test for matching intent action and type. 

I think for the load image option whole code is already implemented. If you try to open a file other than image format ( like mp3 , .txt or mp4 ) using load option it stills open app and show "Error on loading image"

![WhatsApp Image 2021-03-30 at 18 54 05](https://user-images.githubusercontent.com/60041954/112996216-9b7ade80-9189-11eb-91d7-dbaf38e1cea9.jpeg) 
 
I have tried it with Android 10 on Samsung Galazy M01

harsh-hash commented on Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/OraFileIntentTest.java:
```diff
@@ -0,0 +1,135 @@
+package org.catrobat.paintroid.test.espresso;
```
done

harsh-hash commented:
@wslany sir I have added code to load the image. Please review it.

harsh-hash commented:
@wslany sir, please review it. I have added code to load the ora file image.

wslany reviewed (changes requested):
Hi, please fix the lint / static analysis warnings: https://jenkins.catrob.at/job/Paintroid/job/PR-855/14/analysis/

harsh-hash commented:
@wslany sir due to some issues I have pushed some bad commits due to which these warnings just popped up. So I have created a new [PR-208](https://github.com/Catrobat/Paintroid/pull/878). Please have a look at it. thanks

harsh-hash commented on Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/OraFileIntentTest.java:
```diff
@@ -0,0 +1,153 @@
+/*
+ *  Paintroid: An image manipulation application for Android.
+ *  Copyright (C) 2010-2015 The Catrobat Team
```
done

harsh-hash commented:
@wslany I have fixed all warnings except the Regexp error ( Spaces aren't allowed for indentation. Tabs only ). could you tell me how to fix them? As per the error says Spaces aren't allowed for indentation. Tabs only I tried but it's not fixing it.

ThorstenBandel commented:
I guess your Api ( android studio? ) has the indentation set as spaces instead of tabs (and probably on autoconvert) you can change that in the settings and convert the file back

harsh-hash commented:
> I guess your Api ( android studio? ) has the indentation set as spaces instead of tabs (and probably on autoconvert) you can change that in the settings and convert the file back

Thanks, @ThorstenBandel it works. Could you please review it?

wslany commented:
@harsh-hash sorry, could you please resolve the merge conflict?

wslany commented:
Hi, could you please resolve the merge conflicts?

harsh-hash commented:
> Hi, could you please resolve the merge conflicts?

@wslany sorry I will solve it by tomorrow.

harsh-hash commented:
@wslany @ThorstenBandel  sir => #901 Please review it
